src/apis/Event.re

```diff
@@ -1,106 +1,143 @@
+  [@bs.get] [@bs.return nullable]
+  external defaultPrevented: T.target('a) => option(bool) =
+    "defaultPrevented";
+  [@bs.get] external dispatchConfig: T.target('a) => T.r = "dispatchConfig";
+  [@bs.get] [@bs.return nullable]
+  external eventPhase: T.target('a) => option(float) = "eventPhase";
+  [@bs.send] external preventDefault: T.target('a) => unit = "preventDefault";
+  [@bs.send]
+  external isDefaultPrevented: T.target('a) => bool = "isDefaultPrevented";
+  [@bs.send]
+  external stopPropagation: T.target('a) => unit = "stopPropagation";
+  [@bs.send]
+  external isPropagationStopped: T.target('a) => bool =
+    "isPropagationStopped";
+  [@bs.get] [@bs.return nullable]
+  external isTrusted: T.target('a) => option(bool) = "isTrusted";
+  [@bs.get] external nativeEvent: T.target('a) => 'a = "nativeEvent";
+  [@bs.send] external persist: T.target('a) => unit = "persist";
+  [@bs.get] [@bs.return nullable]
+  external target: T.target('a) => option(float) = "target";
+  [@bs.get] external timeStamp: T.target('a) => float = "timeStamp";
+  [@bs.get] [@bs.return nullable]
+  external _type: T.target('a) => option(string) = "type";
 };
 
-type responderSyntheticEvent('a) = {
-  .
-  // synthethicEvent keys
-  "bubbles": Js.Nullable.t(bool),
-  "cancelable": Js.Nullable.t(bool),
-  "currentTarget": float,
-  "defaultPrevented": Js.Nullable.t(bool),
-  "dispatchConfig": {. "registrationName": string},
-  "eventPhase": Js.Nullable.t(float),
-  [@bs.meth] "preventDefault": unit => unit,
-  [@bs.meth] "isDefaultPrevented": unit => bool,
-  [@bs.meth] "stopPropagation": unit => unit,
-  [@bs.meth] "isPropagationStopped": unit => bool,
-  "isTrusted": Js.Nullable.t(bool),
-  "nativeEvent": 'a,
-  [@bs.meth] "persist": unit => unit,
-  "target": Js.Nullable.t(float),
-  "timeStamp": float,
-  "_type": Js.Nullable.t(string),
-  // responderSyntheticEvent additional key
-  "touchHistory": {
-    .
-    "indexOfSingleActiveTouch": float,
-    "mostRecentTimeStamp": float,
-    "numberActiveTouches": float,
-    "touchBank":
-      array({
-        .
-        "touchActive": bool,
-        "startPageX": float,
-        "startPageY": float,
-        "startTimeStamp": float,
-        "currentPageX": float,
-        "currentPageY": float,
-        "currentTimeStamp": float,
-        "previousPageX": float,
-        "previousPageY": float,
-        "previousTimeStamp": float,
-      }),
-  },
+module SyntheticEvent = {
+  type t_('a) = {
+    bubbles: Js.Nullable.t(bool),
+    cancelable: Js.Nullable.t(bool),
+    currentTarget: float,
+    defaultPrevented: Js.Nullable.t(bool),
+    dispatchConfig: registrationName,
+    eventPhase: Js.Nullable.t(float),
+    isTrusted: Js.Nullable.t(bool),
+    nativeEvent: 'a,
+    target: Js.Nullable.t(float),
+    timeStamp: float,
+    [@bs.as "type"]
+    _type: Js.Nullable.t(string),
+  }
+  and registrationName = {registrationName: string};
```

Comment:
Doesn't need dependent definition (`and`)

Reply:
I made the choice to have `t` first, but as you said it's not needed; they are not recursively defined.